documentation/Feedback.md

```diff
@@ -0,0 +1,267 @@
+                components: [actionRow]
+            });
+
+            // Update context for this user in the database
+            await db.db('contrabot').collection("users").updateOne(
+                { userId: user.userId }, 
+                { 
+                    $set: { 
+                        feedbackRequestSent: true
+                    }
+                }
+            );
+        }
+    }
+};
+
+const job = new cron.CronJob('0 0 * * * *', checkForFeedbackRequests); // checks for Feedback every hour
+job.start();
+```
+
+### Feedback Questions
+
+The bot uses an array of predefined questions to query users for feedback. These questions are presented in sequence, starting from the first index in the array. After the user has responded to all questions, they are thanked for their feedback.
+
+The `Feedbackquestions` array is initialized with two questions:
+
+- A general feedback question: 'Gibt es sonst noch etwas, das du uns mitteilen möchtest?'
+- A request for the user's email: 'Was ist deine Mailadresse?'
+
+The function `sendSurveyQuestions` initiates the feedback process by sending the first question from the `Feedbackquestions` array to the user.
+
+**startSurvey.ts**
+```typescript
+const Feedbackquestions = [
+    // General feedback question
+];
+
+async function sendSurveyQuestions(interaction: any) {
+    console.log("Starting survey for user " + interaction.user.id);
+    interaction.user.send(Feedbackquestions[0]);  // Send the first survey question
+}
+
+export { sendSurveyQuestions, Feedbackquestions }
+```
+
+To streamline the user's experience, button interactions are employed. When a user interacts with the 'Start Survey' button, the bot:
+
+1. Recognizes the button interaction.
+2. Defers the update to keep the button's state consistent.
+3. Calls the `sendSurveyQuestions` function to initiate the survey. 
+4. Updates the user's context in the database, marking that the feedback process is underway and recording the current feedback question index.
+
+**index.ts**
+```typescript
+// check if the interaction is a button interaction
+else if (interaction.isButton()) {
+    const buttonId = interaction.customId;
+
+    if (buttonId === 'start_survey') {
+        await interaction.deferUpdate();
+        sendSurveyQuestions(interaction);
+
+        // Update context for this user in the database to indicate feedback process has started
+        await db.db('contrabot').collection("users").updateOne(
+            { userId: interaction.user.id }, 
+            { 
+                $set: { 
+                    feedbackInProgress: true,
+                    currentFeedbackQuestionIndex: 0
+                }
+            }, 
+            { upsert: true }
+        );
+    }
+}
+```
+
+### API Configuration and Feedback Message Handler
+
+This part configures the API for Google Sheets, where feedback responses will be stored.
```

Comment:
Why would we store the feedback on Google's servers, if we can just store it on our own ones?

Reply:
We store the response in a google sheet since it will be easier to work with having a cleaner user interface which will make it possible for non programmers in the longterm to analyse the data